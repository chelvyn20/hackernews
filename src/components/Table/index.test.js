import { createRoot } from 'react-dom/client';
import { act, create } from 'react-test-renderer';
import { Table } from './index';
import { list } from '../../dummy/DummyData';
describe('Table Component', () => {
  const props = {
    list: list,
    onDismiss: jest.fn(),
  };

  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it('should render, show two items in list and  ', () => {
    act(() => {
      createRoot(container).render(<Table {...props} />);
    });
    const tableRow = container.getElementsByClassName('table-row');
    expect(tableRow.length).toBe(2);

    const button = container.getElementsByClassName('button-inline')[0];
    act(() => {
      button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(props.onDismiss).toHaveBeenCalledTimes(1);
  });

  test('snapshots', () => {
    const component = create(<Table {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
