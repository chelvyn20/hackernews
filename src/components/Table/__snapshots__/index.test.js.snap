// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component snapshots 1`] = `
<div
  className="table"
>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href="https://facebook.github.io/react/"
      >
        React
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
       
      Jordan Walke
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      3
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      4
       
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href="https://github.com/reactjs/redux"
      >
        Redux
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
       
      Dan Abramov, Andrew Clark
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      5
       
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
</div>
`;
